pyspeckit/spectrum/units.py

```diff
@@ -665,31 +512,27 @@ def convert_to_unit(self, unit, **kwargs):
         Uses as_unit for the conversion, but changes internal values rather
         than returning them.
         """
-        if unit not in unit_type_dict:
-            raise ValueError("Invalid unit; not in units.unit_type_dict.")
+        if unit is None:
+            unit = u.dimensionless_unscaled
+
+        try:
+            unit = u.Unit(unit)
+        except ValueError:
+            raise ValueError('Invalid unit')
```

Comment:
Won't this raise a `u.UnitsError`?  If not, maybe it can.   This is minor though.